cdragontoolbox/wad.py

```diff
@@ -222,6 +222,17 @@ def set_unknown_paths(self, path):
                 else:
                     wadfile.path = f"{path}/{wadfile.path_hash:016x}"
 
+    def sanitize_paths(self):
+        """Truncate files that have a length of over 255"""
+
+        for wadfile in self.files:
+            path, filename = os.path.split(wadfile.path)
+            if (len(filename) < 256):
+                continue
+
+            basename, ext = os.path.splitext(filename)
+            wadfile.path = os.path.join(path, basename[:255-21] + "." + hex(wadfile.path_hash)[2:] + ext)
```

Comment:
You can use f-strings for the formatting.
Something like that: `f"{basename[:255-21}}.{wadfile.path_hash:016x}{ext}`